addons/web_editor/static/lib/odoo-editor/src/OdooEditor.js

```diff
@@ -328,7 +328,7 @@ export class OdooEditor extends EventTarget {
         // Rollback if node.ouid changed. This ensures that nodes never change
         // unbreakable ancestors.
         node.ouid = node.ouid || getOuid(node, true);
-        if (testunbreak) {
+        if (testunbreak && !(node.nodeType === Node.TEXT_NODE && !node.length)) {
```

Comment:
Is it ok that this only considers really empty text nodes but not whitespace?

Reply:
This seems to be enough to deal with the problems emerging from formatted HTML, and it is also the least invasive check.

So IMO it is OK as long as we don't find other related problems

cc. @Goaman, as he is The One Who Found The Fix. #praise

Reply:
OK. But did you actually test with whitespace? If that doesn't work, wouldn't `isVisibleStr` be our go-to function for this sort of situation?

Reply:
I've added some test to ensure it work properly in all cases ( in the retargeted PR #73541 )